lib/metadata/BucketInfo.js

```diff
@@ -68,6 +102,22 @@ export default class BucketInfo {
                 MfaDelete === 'Enabled' ||
                 MfaDelete === 'Disabled');
         }
+        if (websiteConfiguration) {
+            assert.strictEqual(typeof websiteConfiguration, 'object');
+            const { indexDocument, errorDocument, redirectAllRequestsTo,
+                routingRules } = websiteConfiguration;
+            assert(indexDocument === undefined ||
+                typeof indexDocument === 'string');
+            assert(errorDocument === undefined ||
+                typeof errorDocument === 'string');
+            assert(redirectAllRequestsTo === undefined ||
+                typeof redirectAllRequestsTo === 'object');
+            assert(routingRules === undefined ||
+                Array.isArray(routingRules));
+            websiteConfig = websiteConfiguration instanceof
```

Comment:
Ah, I see. In that case, IMHO, the instance of check should happen before the assertions and use getters instead of accessing the props. directly. Otherwise are just stringifying WebsiteConfiguration instance in BucketInfo.serialize() method, which nullifies the purpose of the class.